src/Microsoft.DotNet.Wpf/src/PresentationCore/MS/internal/WindowsRuntime/Windows/UI/ViewManagement/UISettingsRcw.cs

```diff
@@ -0,0 +1,74 @@
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Security;
+using System.Windows.Media;
+
+namespace MS.Internal.WindowsRuntime
+{
+    namespace Windows.UI.ViewManagement
+    {
+        /// <summary>
+        /// Contains internal RCWs for invoking the UISettings
+        /// </summary>
+        internal static class UISettingsRCW
+        {
+
+            public static object GetUISettingsInstance()
+            {
+                const string typeName = "Windows.UI.ViewManagement.UISettings";
+                IntPtr hstring = IntPtr.Zero;
+                Marshal.ThrowExceptionForHR(NativeMethods.WindowsCreateString(typeName, typeName.Length, out hstring));
```

Comment:
just a tip: using intermediate results such as hr = ... Marshal.ThrowExceptionForHR(hr) makes things easier to step through using a debugger

Reply:
also we should have a marshaller for HSTRING since it was removed from .NET Core, we might need it more and more in the future

Reply:
> just a tip: using intermediate results such as hr = ... Marshal.ThrowExceptionForHR(hr) makes things easier to step through using a debugger

You mean like breaking it in two statements, first we fetch hr and then call ThrowExceptionForHr method ?

Reply:
yes